frontend/lib/framework/Context.cpp

```diff
@@ -639,32 +639,39 @@ void Context::setFilePathForModuleId(ID moduleID, UniqueString path) {
     printf("%i SETTING FILE PATH FOR MODULE %s -> %s\n", queryTraceDepth,
            moduleIdSymbolPath.c_str(), path.c_str());
   }
+  // check that querying the module ID works...
+  UniqueString gotPath;
+  bool ok = filePathForId(moduleID, gotPath);
   #ifndef NDEBUG
-    // check that querying the module ID works...
-    UniqueString gotPath;
-    bool ok = filePathForId(moduleID, gotPath);
     CHPL_ASSERT(ok);
+  #endif
 
-    // ... and gives the same path
+  // ... and gives the same path
 
-    // Note: if this check causes problems in the future, it could
-    // be removed, or we could wire up setFileText used in tests
-    // to work with the LLVM VirtualFilesystem
+  // Note: if this check causes problems in the future, it could
+  // be removed, or we could wire up setFileText used in tests
+  // to work with the LLVM VirtualFilesystem
 #if LLVM_VERSION_MAJOR <= 11
     llvm::SmallVector<char, 64> realPath, realGotPath;
 #else
     llvm::SmallVector<char> realPath, realGotPath;
 #endif
-    std::error_code errPath;
-    std::error_code errGotPath;
-    errPath = llvm::sys::fs::real_path(path.str(), realPath);
-    errGotPath = llvm::sys::fs::real_path(gotPath.str(), realGotPath);
-    if (errPath || errGotPath) {
-      // ignore the check if there were errors
-    } else {
-      CHPL_ASSERT(realPath == realGotPath);
-    }
-  #endif
+  std::error_code errPath;
+  std::error_code errGotPath;
+  errPath = llvm::sys::fs::real_path(path.str(), realPath);
+  errGotPath = llvm::sys::fs::real_path(gotPath.str(), realGotPath);
+  if (!ok || errPath || errGotPath) {
+    // ignore the check if there were errors
+  } else
+    // Check for duplicate modules names, but skip over bundled modules
+    // since we don't necessarily want to preclude the user from using
+    // the same names that we happen to have chosen.
```

Comment:
Usually, Dyno has assertions for things that absolutely must be enforced. You mentioned in the OP that conflicting modules cause one module to override the other -- doesn't this mean that while we want not to issue this error, doing so might lead to incorrect behavior / miscompilation?

Reply:
Potentially.  I was imagining that if we were doing something incorrect or inappropriate here, the existing tests that required the additional check (the ones named in the OP) would fail.  But I admittedly did not look into them at all after getting them passing.

Taking a look at them now, it may very well be that they are outdated or in need of attention (e.g., should they be using one of the `--prepend*` paths to change the standard/internal module paths?  But I also suspect that this aspect of the language design also needs more attention anyway (e.g., how do we prevent the introduction of a new standard or internal module in the future from hijacking a user module if the user's app happened to use the same name), and am not sure I want to take that question on myself in this PR.

Arguably this is a lazy answer and lazy answers are lame, but my goal here was to focus on the thing that broke a user's code and required time from developers to diagnose to prevent it from happening again while not making things worse.  I don't know that I have the time to make other things better as well beyond this at present.

Reply:
> e.g., should they be using one of the --prepend* paths to change the standard/internal module paths?

I definitely think this is the way to go, and we shouldn't rely _only_ on naming to override modules. In the long run, I think we should also warn for bundled paths, and let `--prepend*` take care of finding the _single_ correct module.